### PR TITLE
Services dhcp relay

### PR DESCRIPTION
**Pull request by Bouwdie**

### CODE REVIEW

File: usr/local/www/services_dhcp_relay.php [hunk 1]
```diff
@@ -106,106 +125,73 @@
...
-		document.iform.interface.disabled = 0;
-		document.iform.agentoption.disabled = 0;
-	} else {
-		document.iform.server.disabled = 1;
-		document.iform.interface.disabled = 1;
-		document.iform.agentoption.disabled = 1;
-	}
+if ($dhcpd_enabled)
+{
+	echo "DHCP Server is currently enabled. Cannot enable the DHCP Relay service while the DHCP Server is enabled on any interface.";
```
Comment (SjonHortensius) [inline usr/local/www/services_dhcp_relay.php]: Could you wrap this in an alert? http://getbootstrap.com/components/#alerts

File: usr/local/www/services_dhcp_relay.php [hunk 2]
```diff
@@ -106,106 +125,73 @@
...
+if ($input_errors)
+	print_input_errors($input_errors);
+
+if ($savemsg)
+	print_info_box($savemsg);
+
+$form = new Form;
+
+$section = new Form_Section('DHCP Relay configuration');
+$form->add($section);
```
Comment (SjonHortensius) [inline usr/local/www/services_dhcp_relay.php]: I'm a bit surprised this works, can you move the add to be after all add() calls have been done? Eg. just before `print $form`

File: usr/local/www/services_dhcp_relay.php (hunk 1 above)
Comment (Bouwdie) [inline usr/local/www/services_dhcp_relay.php]: I was thinking about it, wrapped in in the print_error_messages first. Maybe the messaging system can be abstracted.

File: usr/local/www/services_dhcp_relay.php [hunk 3]
```diff
@@ -40,13 +40,29 @@
 ##|-PRIV
 
 require("guiconfig.inc");
+require('classes/Form.class.php');
+
+function filterDestinationServers(array $destinationServers)
+{
+	return array_unique(
+		array_filter(
```
Comment (SjonHortensius) [inline usr/local/www/services_dhcp_relay.php]: I actually think if you're using array_filter with (!empty), you can remove the callback entirely, as the default of array_filter is remove everything that evals to false (which all empties do)

File: usr/local/www/services_dhcp_relay.php (hunk 2 above)
Comment (Bouwdie) [inline usr/local/www/services_dhcp_relay.php]: Why wouldn't it work? You work on the same object reference. For clarity i shall move it down.

**Comment by Bouwdie** [comment]
I see that services_dhcpv6_relay.php is almost the same as this form. Do we combine these or keep duplicated forms?

**Comment by SjonHortensius** [comment]
You can definitely combine them, just check if/where the v6 variant is
referenced in existing files

**Pull request merged**